npipe_windows.go

```diff
@@ -443,6 +446,12 @@ func (c *PipeConn) Write(b []byte) (int, error) {
 
 // Close closes the connection.
 func (c *PipeConn) Close() error {
+	c.closedMutex.Lock()
+	defer c.closedMutex.Unlock()
+	if c.closed {
+		return nil
+	}
+	c.closed = true
 	return syscall.CloseHandle(c.handle)
```

Comment:
I don't think it matters really, just making the observation. Too bad you can't do an early defer or something in Go.